compiler/rustc_infer/src/infer/error_reporting/note_and_explain.rs

```diff
@@ -32,6 +32,12 @@ impl<'tcx> TypeErrCtxt<'_, 'tcx> {
                         diag.note("no two closures, even if identical, have the same type");
                         diag.help("consider boxing your closure and/or using it as a trait object");
                     }
+                    (ty::Coroutine(def_id1, ..), ty::Coroutine(def_id2, ..))
+                        if self.tcx.coroutine_is_async(def_id1)
+                            && self.tcx.coroutine_is_async(def_id2) =>
+                    {
+                        diag.note("every async block has a different type even when the output type is the same");
```

Comment:
Do suggest any changes in the note's text that could make it clearer or shorter.